packages/overlay/src/integrations/sentry/context/SearchContext.tsx

```diff
@@ -1,22 +1,43 @@
-import { createContext, ReactNode, useContext, useState } from 'react';
-
+import { createContext, ReactNode, useCallback, useContext, useState } from 'react';
+import { Span } from '../types';
 type SearchContextType = {
   query: string;
   setQuery: (query: string) => void;
+  showOnlyMatched: boolean;
+  setShowOnlyMatched: (showOnlyMatched: boolean) => void;
 };
 
 const SearchContext = createContext<SearchContextType | undefined>(undefined);
 
 export function SearchProvider({ children }: { children: ReactNode }) {
   const [query, setQuery] = useState('');
+  const [showOnlyMatched, setShowOnlyMatched] = useState(false);
 
-  return <SearchContext.Provider value={{ query, setQuery }}>{children}</SearchContext.Provider>;
+  return (
+    <SearchContext.Provider value={{ query, setQuery, showOnlyMatched, setShowOnlyMatched }}>
+      {children}
+    </SearchContext.Provider>
+  );
 }
 
 export function useSearch() {
   const context = useContext(SearchContext);
   if (context === undefined) {
     throw new Error('useSearch must be used within a SearchProvider');
   }
-  return context;
+
+  const matchesQuery = useCallback(
+    (span: Span): boolean => {
+      const q = context.query.toLowerCase();
+      return (
+        (span.span_id.toLowerCase().includes(q) ||
+          span.op?.toLowerCase().includes(q) ||
+          span.description?.toLowerCase().includes(q)) ??
+        false
```

Comment:
I think it would be more efficient if we create a:

```js
const queryMatcher = new RegExp(RegExp.escape(query), 'i');
```

and then do `queryMatcher.test(span.op)` etc. afterwards. You may need to polyfill [`RegExp.escape`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/escape) as it doesn't seem to have wide browser support without flags.

Finally, a bit of a nitpick, we may wanna limit matching to only when the query is more than 2 characters as I don't want to match on `a` or even `ab` probably.